SDK/AppCenterCrashes/Microsoft.AppCenter.Crashes.Windows.Shared/Crashes.cs

```diff
@@ -352,5 +372,70 @@ private Task SendErrorAttachmentsAsync(Guid errorId, IEnumerable<ErrorAttachment
             }
             return Task.WhenAll(tasks);
         }
+
+        private ErrorReport BuildErrorReport(ManagedErrorLog log)
+        {
+            if (_errorReportCache.ContainsKey(log.Id))
+            {
+                return _errorReportCache[log.Id];
+            }
+            var file = ErrorLogHelper.GetStoredExceptionFile(log.Id);
+            if (file == null)
+            {
+                return null;
+            }
+            var exception = ErrorLogHelper.ReadExceptionFile(file);
+            var report = new ErrorReport(log, exception);
+            _errorReportCache.Add(log.Id, report);
+            return report;
+        }
+
+        private void ChannelSendingLog(object sender, SendingLogEventArgs e)
+        {
+            var report = ProcessEventHandlers(e, false);
+            if (report != null)
+            {
+                SendingErrorReport?.Invoke(sender, new SendingErrorReportEventArgs { Report = report });
+            }
+        }
+
+        private void ChannelSentLog(object sender, SentLogEventArgs e)
+        {
+            var report = ProcessEventHandlers(e);
+            if (report != null)
+            {
+                SentErrorReport?.Invoke(sender, new SentErrorReportEventArgs { Report = report });
+            }
+        }
+
+        private void ChannelFailedToSendLog(object sender, FailedToSendLogEventArgs e)
+        {
+            var report = ProcessEventHandlers(e);
+            if (report != null)
+            {
+                FailedToSendErrorReport?.Invoke(sender, new FailedToSendErrorReportEventArgs { Report = report, Exception = e.Exception });
+            }
+        }
+
+        private ErrorReport ProcessEventHandlers(ChannelEventArgs e, bool deleteExceptionFile = true)
```

Comment:
Can you please name this method more descriptively? How is it processing an event handler?

Reply:
I guess it's processing event args? But still that would be a slightly confusing name to me

Reply:
Maybe we could move the deletion of the exception file outside the method implementation, and rename this to `TryBuildErrorReport` or something

Reply:
That would duplicate the if logic to move the delete outside, do you have another suggestion instead?

Reply:
I refactored.